department-of-veterans-affairs/vets-api
# [56520] Monkeypatching StatsD::Instrument functions to replace StatsD key strings, dots with underscores

## Summary

- This PR adds monkey patching to the `StatsD::Instrument` class so that any dots `.` in the `StatsD` key are replaced with underscores `_`. This is done to preserve historical functionality with many dashboard and monitoring metrics

- Unfortunately, we have to disable the monkey patch class on test environments, as we use test functions from the `StatsD` gem, such as `trigger_statsd_increment`, which test for `StatsD` calls separate from the monkey patched class

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#56520


## Testing done

- Run rails server and perform various actions 
- Inspect `log/statsd.log` on localhost
- Confirm that all `StatsD` keys are made up of underscores `_` separating tokens instead of dots `.`


## What areas of the site does it impact?
Metrics and Monitoring

## Acceptance criteria

- [ ] Confirm that any `StatsD` metric you use has underscores `_` instead of dots `.`
- [ ] Modules are weird, could be good to distinctly confirm that any relevant modules also have taken these changes (I added copies of the monkey patch to the different modules initializers that have `StatsD` in the existing initializer)
